service/history/historyEngine.go

```diff
@@ -1604,6 +1610,7 @@ Update_History_Loop:
 						return nil, err
 					}
 					if failWorkflow {
+						isComplete = true
```

Comment:
was this a bug that existed before ? I don't follow how the rest of the changes are related to this

Reply:
This is a bug that existed. It should have been a separate PR.